pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -104,10 +104,20 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 	log.Info("Maintaining Shoot")
 
 	var (
-		maintainedShoot             = shoot.DeepCopy()
-		operations                  []string
-		reasonForImageUpdatePerPool []string
-		err                         error
+		maintainedShoot = shoot.DeepCopy()
+		// for maintenance operations unrelated to machine images and Kubernetes versions
+		operations []string
+
+		workerToReasonForKubernetesUpdate map[string]string
+		workerToReasonForImageUpdate      map[string]string
+
+		// Note: the updates might still be rejected by APIServer validation. This simply means that we found a suitable version to update to.
+		workerToSuccessfulImageUpdate      map[string]string
+		workerToSuccessfulKubernetesUpdate map[string]string
+
+		workerToFailedKubernetesUpdate map[string]string
+		workerToFailedImageUpdate      map[string]string
```

Comment:
It gives the feeling that we try to simulate an object with these maps - like every map holds the same key (worker name) and holds one field of that object.

Maybe it can be improved with introducing a new struct?

```go
type updateResult struct {
	description   string
	reason        string
	isSuccessfull bool
}
```

And then you would have only 2 maps
```go
	var workerToKubernetesUpdate map[string]updateResult
	var workerToMachineImageUpdate map[string]updateResult
```

WDYT?